### ISSUE TITLE
Bug in tf.keras bidirectional LSTM when time_major is true

### ISSUE THREAD

**Issue by sonu1-p** [tensorflow/tensorflow#39635]
**System information**
- Have I written custom code: Yes
- OS Platform and Distribution: Macos Mojave 10.14.6
- TensorFlow installed from: Binary
- TensorFlow version: v2.2.0-rc4-8-g2b96f3662b 2.2.0
- Python version: 3.6.3

**Bug description**

When using bidirectional layer with forward/backward lstms with time_major=True and merge mode concat(same issue exists in other modes too), it produces incorrect results due to the below line:
https://github.com/tensorflow/tensorflow/blob/2b96f3662bd776e277f86997659e61046b56c315/tensorflow/python/keras/layers/wrappers.py#L658
If time_major=true (the input shape of bidi lstm is - [seq_len, batch_size, hidden_size]), axis 1 represents batch dimesnion and we end up reversing y_rev in batch dimension before concatenation while it should have been reversed in the dimension representing seq_len(axis 0).

This works fine when LSTM is time_major=False as in that case axis 1 represents seq_len.  Ideally it should see which axis - axis 0 or axis 1 represent the time dimension and reverse on that axis, instead of generically reversing on axis 1.

Could you please fix this bug as the time_major version of the lstms is more efficient.

**Standalone code to reproduce the issue**
```
import tensorflow as tf
import numpy as np

seq_len = 2
batch_size = 1
feature_dim = 1

input = tf.keras.Input(shape=(seq_len, feature_dim))
# Transpose input to be time major
input_transposed = tf.transpose(input, perm=[1,0,2])
output = tf.keras.layers.Bidirectional(tf.keras.layers.LSTM(1, return_sequences=True, time_major=True), name='bi')(input_transposed)
model = tf.keras.Model(inputs=input, outputs=output)

# Set all the weights to be one for simplicity
rnn_layer = model.get_layer('bi')
weights = rnn_layer.get_weights()
new_w = [np.ones(x, dtype=np.float32) for x in [(feature_dim, 4), (1, 4), (4)] * 2]
rnn_layer.set_weights(new_w)

model.save("test.h5")
x = np.ones((batch_size, seq_len, feature_dim), dtype=np.float32)
expected = model.predict(x)
print(expected)
```

Expected result is :
```
[[[0.6082834  0.87263733]    
  [0.87263733 0.6082834 ]]]
which is:
[[[ forward_layer_seq_1, backward_layer_seq_1 ]]
 [[ forward_layer_seq_2, backward_layer_seq_2]]]
But what we get is:
[[[0.6082834  0.6082834 ]]
 [[0.87263733 0.87263733]]]
[[[ forward_layer_seq_1, backward_layer_seq_2 ]]
 [[ forward_layer_seq_2, backward_layer_seq_1]]]
```

**Comment by ravikyram** [comment]
@sonu1-p 

I have tried in colab with TF nightly version (2.3.0-dev20200518) and i am not seeing any issue in nightly version.Please, find the gist [here](https://colab.sandbox.google.com/gist/ravikyram/ddebb9a159077acb4072d6a8aad25cab/untitled899.ipynb).You could use tf-nightly for now and in the next couple of months new stable version will be released.Please, close this thread if your issue got resolved. Thanks!

**Comment by sonu1-p** [comment]
Yes I see that its fixed in master branch:
https://github.com/tensorflow/tensorflow/blob/ef45324fc62fc9a911e5771a40f9790900500de9/tensorflow/python/keras/layers/wrappers.py#L656

Looks like it got fixed by this commit last month - https://github.com/tensorflow/tensorflow/commit/1a09bbf34ea07713a66cc4385800ab2744587884

I am also not able to reproduce it with tf-nightly. Thanks will close this.
